app/services/learning_units/resource_sorter_service.rb

```diff
@@ -0,0 +1,25 @@
+module LearningUnits
+  class ResourceSorterService
+    delegate :name, to: :learning_unit, prefix: :learning_unit
+    def initialize(learning_unit)
+      @learning_unit = learning_unit
+    end
+
+    def sorted_resources
```

Comment:
This method could be replaced with resources.flatten.each_slice(3).to_a.
However, I don't know which is clearer. 

Also, iterate over the same array two times could be wasteful (first on the service and then in the view).  However, I don't think it is worth to try to optimize at this level.